### ISSUE TITLE
Filtering a listing numerically fails non-deterministically

### ISSUE THREAD

**Issue by MattF-NSIDC** [quarto-dev/quarto-cli#4606]
### Bug description

# Reproducing

To set up for reproduction quickly, clone this repo and follow the instructions in "Repro steps" below:

https://github.com/MattF-NSIDC/sscce-quarto-listing-issue

## Setup steps

1. Create the project
   ```
   quarto create-project demo-filter-issue --type=website
   ```
1. Change the `index.qmd` to a "table"-type listing
   ```
   ---
   title: "demo-filter-issue"
   listing: "table"
   ---
   ```
1. Create an `items/` dir and populate with 4 markdown files with numbered titles, e.g. `# Item #1: Foo`, `# Item #2: Bar`, etc.
1. Remove `about.qmd`

## Repro steps

1. `quarto preview`
1. In your browser on the index page, try filtering the listing for `1`, `2`, `3`, and `4` separately.
1. At least one of these filter attempts will not actually cause any filtering. If this doesn't happen, delete `.quarto` and `_site` directories and start over. Each time, the behavior should be different. E.g. the first time filtering for `1` may be ineffective, and next time filtering for `4` may be ineffective, and the next time both `1` and `2` will not be filterable.

## Repro screenshot sample

Because this issue is non-deterministic, each run of `quarto preview` will behave differently. Therefore, the issue won't reproduce exactly the same as these screenshots.

* Filtering for `2` is effective

![image](https://user-images.githubusercontent.com/18539827/222514096-655f0b9d-7913-45bd-8836-a98e7d7d9a1c.png)

* Filtering for `3` is ineffective

![image](https://user-images.githubusercontent.com/18539827/222514118-bda7845b-24ca-4bba-b76c-9c9742a9a213.png)

# Versions tested

* Quarto 1.3.241, 1.3.217, 1.2.335
* Ubuntu 20
* No IDE

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by MattF-NSIDC** [comment]
Additional observations:


* Filtering for # always returns no results.
* Only numeric searches are affected in this non-deterministic way. E.g. if filtering for `2` is ineffective, filtering for `bar` will be effective.

**Comment by dragonstyle** [comment]
Great one! I've found the underlying issue which seems to only affect table style listings (at least the number indeterminate behavior). It should be fixed in our next pre-release build (7d221f48a55f9f0085ea34fd72a6f2794c5a7a0c).

Currently, it appears search/filter doesn't support elements like `#` - there is open issue in the parent repo here:

https://github.com/javve/list.js/issues/699

Thanks for reporting this- was fun to track down!

**Comment by MattF-NSIDC** [comment]
>the href would often contain port numbers, making the behavior appear random.

That behavior makes so much sense now :laughing:

Nice find!
